### DISCUSSION TITLE
Ruffle graphics hardware requirements

### DISCUSSION THREAD

**Discussion by colmde** [ruffle-rs/ruffle#1530] [General]
When I run Ruffle I get the error...

Ruffle requires hardware acceleration, but no compatible graphics device was found supporting Vulkan, DirectX 12, Metal or Web GPU

Why does it need this only when run from the command line? I can run the demo on the website with absolutely no problems! [https://ruffle.rs/demo/](https://ruffle.rs/demo/) - why doesn't need hardware accelaration there?

**Comment by Herschel** [comment]
The desktop build currently uses [wgpu-rs](https://github.com/gfx-rs/wgpu-rs) which requires one of the above rendering APIs to be available (Vulkan, DX12, Metal, ...).  Unfortunately OpenGL is not well supported by wgpu-rs yet, so Ruffle desktop currently doesn't run on systems that only have OpenGL available. I would like to add a software rendering backend at some point, also.

The web build uses WebGL when available, and falls back to using HTML5 canvas otherwise, which allows it to run without hardware acceleration.
